src/controller_examples/simple_controller/spec/custom_resource.rs

```diff
@@ -133,15 +141,56 @@ impl ResourceView for CustomResourceView {
         }
     }
 
-    open spec fn from_dynamic_object(obj: DynamicObjectView) -> CustomResourceView {
-        CustomResourceView {
-            metadata: obj.metadata,
-            spec: CustomResourceSpecView{
-                content: obj.data.get_Object_0()[spec_field()].get_Object_0()[spec_content_field()].get_String_0(),
-            },
-            status: if obj.data.get_Object_0()[status_field()].is_Null() {Option::None} else {Option::Some(CustomResourceStatusView{
-                echoed_content: obj.data.get_Object_0()[status_field()].get_Object_0()[status_echoed_content_field()].get_String_0(),
-            })},
+    open spec fn from_dynamic_object(obj: DynamicObjectView) -> Result<CustomResourceView, ParseDynamicObjectError> {
+        if obj.data.is_Object() {
+            let obj_data = obj.data.get_Object_0();
+            if obj_data.dom().contains(spec_field()) && obj_data.dom().contains(status_field()) {
+                let data_spec = obj_data[spec_field()];
+                let data_status = obj_data[status_field()];
+                if data_spec.is_Object() && (data_status.is_Null() || data_status.is_Object()) {
+                    let obj_data_spec = data_spec.get_Object_0();
+                    if obj_data_spec.dom().contains(spec_content_field()) {
+                        let data_spec_content = obj_data_spec[spec_content_field()];
+                        if data_spec_content.is_String() {
+                            if data_status.is_Null() {
+                                let res = CustomResourceView {
+                                    metadata: obj.metadata,
+                                    spec: CustomResourceSpecView { content: data_spec_content.get_String_0(), },
```

Comment:
Instead of parsing `obj_data_spec` here, probably checking `obj_data_spec.unmarshal().is_Ok()` and then `spec: obj_data_spec.unmarshal().get_Ok()` is better? You can just have one error type called UnmarshalError for the not OK branch of unmarshal.